src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -447,7 +448,10 @@ class ObservationService(
             zone.plantingSubzones
                 .flatMap { subzone ->
                   subzone.monitoringPlots.filter {
-                    it.id in newPermanentPlotIds &&
+                    it.isAvailable &&
+                        it.id !in existingPermanentPlotIds &&
+                        it.permanentCluster != null &&
+                        it.permanentCluster <= zone.numPermanentClusters &&
```

Comment:
This is a bit hard to grok, but the val name is very specific, which helps a lot here. I wonder if the val name should be updated since the underlying filter condition has changed.

Reply:
Updated the name and also broke the expression up into pieces with hopefully meaningful names.